pallets/providers/src/utils.rs

```diff
@@ -212,12 +212,94 @@ where
         Ok(())
     }
 
-    pub fn do_msp_sign_off(_who: &T::AccountId) -> DispatchResult {
-        todo!()
+    pub fn do_msp_sign_off(who: &T::AccountId) -> DispatchResult {
+        // Check that the signer is registered as a MSP and get its info
+        let msp_id =
+            AccountIdToMainStorageProviderId::<T>::get(who).ok_or(Error::<T>::NotRegistered)?;
+
+        let msp = MainStorageProviders::<T>::get(&msp_id).ok_or(Error::<T>::NotRegistered)?;
```

Comment:
I'd use the `expect_or_err` macro here. This is never supposed to happen if you previously found its `msp_id`. I'd assign it a special error, so that if this ever happens, we know that there is an inconsistency in the on-chain storage.